### DISCUSSION TITLE
Workflow model source loading

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#685] [Q&A] [answered]
Hi, I know that the workflow uses the null GModel factory implementation, but I'm wondering where the handoff from the .wf file happens. Where in the workflow example code or in the GLSP server code does the server get the .wf file and turn it into a GModel?

**Comment by tortmayr** [comment]
Hi @gee4vee ,
This is typically the responsibility of the `SourceModelStorage`. In the case of the workflow example,
the Workflow Diagram Module extends the [GModelDiagramModule](https://github.com/eclipse-glsp/glsp-server/blob/132e66d39b58b5de5ee359f9892fc35fce9e5800/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/gmodel/GModelDiagramModule.java#L46). This class binds the [´GModelStorage`](https://github.com/eclipse-glsp/glsp-server/blob/132e66d39b58b5de5ee359f9892fc35fce9e5800/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/gmodel/GModelStorage.java#L69) which can be used to load & read the GModel from a JSON file.

**Comment by gee4vee** [comment]
@tortmayr sorry I should have been clearer. We are looking at the node server implementation only. Can you point me to the relevant code there?

**Comment by tortmayr** [comment, marked as answer]
Sure, for the node implementation the `GModelStorage` code can be found here:
https://github.com/eclipse-glsp/glsp-server-node/blob/b178eb2cf7509fdfa3349e012b33c651340dc0bb/packages/server-node/src/gmodel-lib/gmodel-storage.ts#L42
